### PR TITLE
Add LangVersion for C# 7.3

### PR DESCRIPTION
**Pull request by jcouv**

### CODE REVIEW

**Comment by jcouv** [comment]
@jaredpar @dotnet/roslyn-infrastructure  I see that the [generator](https://ci.dot.net/job/dotnet_dotnet-ci_generator/) job has [run](https://ci.dot.net/job/dotnet_dotnet-ci_generator/307/), but I'm still not seeing CI legs here.
Anything else needed?

**Comment by jcouv** [comment]
@dotnet/roslyn-compiler for review. Adding LangVersion 7.3 to `features/compiler` branch. Thanks

File: src/Compilers/CSharp/Portable/LanguageVersion.cs [hunk 1]
```diff
@@ -80,6 +80,11 @@ public enum LanguageVersion
         CSharp7_2 = 702,
 
         /// <summary>
+        /// C# language version 7.3
+        /// </summary>
+        CSharp7_3 = 703,
```
Comment (jcouv) [inline src/Compilers/CSharp/Portable/LanguageVersion.cs]: :memo: Filed https://github.com/dotnet/roslyn/issues/22957 to remind to add the drop-down entry (in project-system).

**Comment by jcouv** [comment]
@dotnet/roslyn-compiler for review. This would help another PR move forward without gymnastics (https://github.com/dotnet/roslyn/pull/22664 depends on having the new LanguageVersion). Thanks

File: src/Compilers/CSharp/Portable/Errors/ErrorCode.cs [hunk 2]
```diff
@@ -1540,5 +1540,11 @@ internal enum ErrorCode
         #endregion diagnostics introduced for `ref readonly`, `ref ternary` and `ref-like` features in C# 7.2
 
         ERR_PredefinedValueTupleTypeAmbiguous3 = 8356,
+
+        #region diagnostics introduced for C# 7.3
+        ERR_FeatureNotAvailableInVersion7_3 = 8360,
+        #endregion diagnostics introduced for C# 7.3
+
+        // Note: you will need to re-generate compiler code after touching entries that are not ERR_* (build\scripts\generate-compiler-code.cmd)
```
Comment (cston) [inline src/Compilers/CSharp/Portable/Errors/ErrorCode.cs]: Minor point: perhaps `... after adding warnings (build\scripts\...)`.

**Pull request merged**